backend/src/graphql/resolver/util/findUserByIdentifier.ts

```diff
@@ -0,0 +1,35 @@
+import { User as DbUser } from '@entity/User'
+import { UserContact as DbUserContact } from '@entity/UserContact'
+import LogError from '@/server/LogError'
+import { validate, version } from 'uuid'
+
+export const findUserByIdentifier = async (identifier: string): Promise<DbUser | null> => {
+  let user: DbUser | undefined
+  if (validate(identifier) && version(identifier) === 4) {
+    user = await DbUser.findOne({ where: { gradidoID: identifier }, relations: ['emailContact'] })
+    if (!user) {
+      throw new LogError('No user found to given identifier', identifier)
+    }
+  } else if (/^.{2,}@.{2,}\..{2,}$/.exec(identifier)) {
+    const userContact = await DbUserContact.findOne(
+      {
+        email: identifier,
+        emailChecked: true,
+      },
+      { relations: ['user'] },
+    )
+    if (!userContact) {
+      throw new LogError('No user with this credentials', identifier)
+    }
+    if (!userContact.user) {
+      throw new LogError('No user to given contact', identifier)
+    }
+    user = userContact.user
+    user.emailContact = userContact
+  } else {
+    // last is alias when implemented
+    throw new LogError('Unknown identifier type', identifier)
+  }
+
+  return user
+}
```

Comment:
```suggestion
export const findUserByIdentifier = async (identifier: string): Promise<DbUser> => {
  // I would love to see a comment here what each if does, it would help my future self to read this
  // uuid identifier
  if (validate(identifier) && version(identifier) === 4) {
    const user = await DbUser.findOne({ where: { gradidoID: identifier }, relations: ['emailContact'] })
    if (!user) {
      throw new LogError('No user found to given identifier', identifier)
    }
    // this exposes the user email!
    return user
  // I would love to see a comment here what each if does, it would help my future self to read this
  // email identifier
  } else if (/^.{2,}@.{2,}\..{2,}$/.exec(identifier)) {
    const userContact = await DbUserContact.findOne(
      {
        email: identifier,
        emailChecked: true,
      },
      { relations: ['user'] },
    )
    if (!userContact) {
      throw new LogError('No user with this credentials', identifier)
    }
    if (!userContact.user) {
      throw new LogError('No user to given contact', identifier)
    }
    return userContact.user
    // this like would expose the users email - since its know for the user, since he queried this user by email, its not very dangerous, but lets leave out the email completely? Or is this to fast?
    // user.emailContact = userContact
  }

  throw new LogError('Unknown identifier type', identifier)
}
```